### ISSUE TITLE
TypeError: No matching signature found

### ISSUE THREAD

**Issue by riccardosamperna** [DIAGNijmegen/Report-Guided-Annotation#2]
The function ndimage.label:
https://github.com/DIAGNijmegen/Report-Guided-Annotation/blob/d247ee96548f0f72fa7e788524d2348ffed7e292/src/report_guided_annotation/extract_lesion_candidates.py#L29

throws a "TypeError: No matching signature found" when passing a softmax array of type float16.

```
~/.local/lib/python3.8/site-packages/scipy/ndimage/_measurements.py in label(input, structure, output)
    217 
    218     try:
--> 219         max_label = _ni_label._label(input, structure, output)
    220     except _ni_label.NeedMoreBits as e:
    221         # Make another attempt with enough bits, then try to cast to the
```

I solved the problem by casting the softmax numpy array .astype(float).

You should either catch the error with a better error message e.g. "extract_lesion_candidates function only accepts float32/64 numpy arrays" or cast automatically the array to float32/64 before passing it to ndimage.label.

**Comment by joeranbosma** [comment]
Hi @riccardosamperna,

Thanks for raising this issue! Support for float16 inputs is intended, so it should be fixed rather than throwing an error.

Doing ‘.astype(float)’ converts the input to a 64bit float, which isn't optimal when dealing with large inputs. When I'm back in office I can check for a clean way to convert non-supported dtypes, and keep supported dtypes as-is. 

If you could make a PR with this I'm more than happy to merge it in! This also properly attributes your contribution. 

Kind regards,
Joeran

**Comment by joeranbosma** [comment]
This issue is now resolved: see PR #3
